src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -10639,6 +10639,96 @@ namespace WaterThermalTanks {
         return PLRResidualHPWH;
     }
 
+    bool SourceHeatNeed(int const WaterThermalTankNum,
+                        Real64 const OutletTemp,
+                        Real64 const DeadBandTemp,
+                        Real64 const SetPointTemp
+        ){
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Yueyue Zhou
+        //       DATE WRITTEN   May 2019
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS FUNCTION:
+        // Determine by tank type, tank temperature and control mode if source side flow is needed
+
+        // METHODOLOGY EMPLOYED:
+        // <description>
+
+        // REFERENCES:
+        // na
+
+        // Using/Aliasing
+        using ScheduleManager::GetCurrentScheduleValue;
+
+        //return value initialization
+        bool NeedsHeatOrCool = false;
+
+        Real64 AltSetpointTemp;
```

Comment:
These can be defined at the point of usage since they are only used within that scope, and also declared `const` for clarity.